src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,196 @@
 package core.basesyntax;
 
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int CAPACITY_DELTA = 2;
+    private int capacity;
+    private int size;
+    private Node<K, V>[] buckets;
+    private double threshold;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        buckets = new Node[DEFAULT_CAPACITY];
+        threshold = capacity * LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        int index = key != null ? Math.abs(key.hashCode() % capacity) : 0;
```

Comment:
Let's create separate method and use it here and in resize() 
```suggestion
        putInTable(table, key, value);
```

Reply:
Where 
```
private void putInTable(Node<K, V>[] table, K key, V value) {
        int index = calculateIndex(key);
        Node<K, V> current = table[index];
        Node<K, V> newNode = new Node<>(key, value, null);
        if (current == null) {
            table[index] = newNode;
        } else {
            Node<K, V> prev = current;
            while (current != null) {
                if (Objects.equals(newNode.key, current.key)) {
                    current.value = newNode.value;
                    return;
                }
                prev = current;
                current = current.next;
            }
            prev.next = newNode;
        }
        size++;
    }
```